array.h

```diff
@@ -20,10 +20,19 @@
 #define NDARRAY_ARRAY_H
 
 #include <array>
+// TODO(jiawen): CUDA *should* support assert on device. This might be due to the fact that we are
```

Comment:
Hmm, I was going to suggest putting this in the test, but I guess that isn't actually possible :(. I think what you have here is fine for now.